docs/usage.md

```diff
@@ -210,6 +210,119 @@ navigation.overrides = {
   }
 }
 ```
+
+# Tree and Partial Tree Insertion & Registering
+
+LRUD supports the ability to register an entire tree at once.
+
+```js
+const instance = new Lrud();
+const tree = {
+  root: {
+    orientation: 'horizontal',
+    children: {
+      alpha: { isFocusable: true },
+      beta: { isFocusable: true },
+      charlie: { isFocusable: true },
+    }
+  }
+}
+
+instance.registerTree(tree);
+// `instance` now has the above tree registered, and has correctly setup active children, indexes, etc. 
+```
+
+## `insertTree()` and nested tree registration
+
+LRUD also supports the ability to register a tree/insert a tree into an already existing branch.
+
+If no parent is given on the top level node of the passed tree, the tree will be inserted under the root node, as per standard `registerNode()` behaviour.
+
+Otherwise, if a `parent` _is_ given, the tree will be inserted under that parent.
+
+### Inserting a tree under the root node
+```js
+const instance = new Lrud();
+instance
+  .registerNode('root', { orientation: 'horizontal' })
+  .registerNode('alpha', { isFocusable: true })
+  .registerNode('beta', { isFocusable: true })
+
+const tree = {
+  charlie: {
+    orientation: 'vertical',
+    children: {
+      charlie_1: { isFocusable: true },
+      charlie_2: { isFocusable: true },
+    }
+  }
+}
+instance.registerTree(tree);
+/*
+the full tree of `instance` now looks like:
+{
+  root: {
+    orientation: 'horizontal',
+    children: {
+      alpha: { isFocusable: true }
+      beta: { isFocusable: true }
+      charlie: {
+        orientation: 'vertical'
+        children: {
+          charlie_1: { isFocusable: true }
+          charlie_2: { isFocusable: true }
+        }
+      }
+    }
+  }
+}
+*/
+```
+
+### Inserting a tree under a specified branch
+
+```js
+const instance = new Lrud();
+instance
+  .registerNode('root', { orientation: 'horizontal' })
+  .registerNode('alpha', { isFocusable: true })
+  .registerNode('beta', { orientation: 'vertical' })
+
+const tree = {
+  charlie: {
+    orientation: 'vertical',
+    parent: 'beta',
```

Comment:
how deep go to search and ho about duplicated key?

Reply:
Good point. Perhaps `parent` needs to be a path...

Reply:
Me and @thomgray have discussed about it offline. We accept that the tree doesn't support duplicated nodes name.

Reply:
None of the IDs at the minute can be duplicate, and `parent` here will dig as far into the tree as it needs to. This is the same as if you were calling `registerNode()` and giving that a parent - it handles any depth.

Happy to talk through in more detail though, and as I said to Emilian earlier, should probably write up some ADRs on what we talk about.